Add Portfolio Works Section

BenjaminSemah opened the pull request.

BenjaminSemah committed:
Add html for Works Section

BenjaminSemah committed:
Add style to my abilities section.

BenjaminSemah committed:
Add style to profile section

BenjaminSemah committed:
Add styles to Works section

BenjaminSemah committed:
Update README.md file

I have updated the app screenshot in the README.md file to capture the updates made.

BenjaminSemah committed:
Update stylesheet and markup for Homepage

BenjaminSemah committed:
Fix linters errors for stylesheet for Homepage

BenjaminSemah committed:
Update skills section

BenjaminSemah commented:
**Add Portfolio Works Section**

I have worked on the following;
- Added languages, frameworks and skills.
- Added social media icons.
- Added four recent works.

moise10r commented on style.css:
```diff
@@ -67,3 +63,184 @@ header {
+  color: #dcdce8;
+}
+
+#project1 {
+  background-image: url(./assets/pro1image.svg);
+  background-repeat: no-repeat;
+  background-position: left;
+  padding: 20px;
+}
+
+#project2 {
+  background-image: url(./assets/pro2image.svg);
+  background-repeat: no-repeat;
+  background-position: right;
+  padding: 20px;
+}
+
+#project3 {
+  background-image: url(./assets/pro3image.svg);
+  background-repeat: no-repeat;
+  background-position: left;
+  padding: 20px;
+}
+
+#project4 {
+  background-image: url(./assets/pro4image.svg);
+  background-repeat: no-repeat;
+  background-position: right;
+  padding: 20px;
+}
```
- [ ] Great job while implementing these projects container. They should align the items inside of it by using a flexbox. Can you please use the CSS flexbox to lay out the contents of these projects?
   - They need to have something like this:
   ```CSS
  #project {
     display: flex;
     ...
   }
   ```

moise10r commented on Index.html:
```diff
@@ -28,6 +28,204 @@ <h2 id="myTitle">Software Developer</h2>
+        <img class="line" src="./assets/vectorLine.svg" alt="horizontal line">
+    </section>
+    <section class="socials">
+        <ul class="profiles">
+            <li>
+                <a href="#">
+                    <img id="github" src="./assets/github.svg" alt="GitHub Logo">
+                </a>
+            </li>
+            <li>
+                <a href="#">
+                    <img id="linkedIn" src="/assets/linkedinn.png" alt="LinkedIn Logo">
+                </a>
+            </li>
+            <li>
+                <a href="#">
+                    <img id="angellist" src="/assets/angellist.svg" alt="Angellist Logo">
+                </a>
+            </li>
+            <li>
+                <a href="#">
+                    <img id="twitter" src="/assets/Twitter.svg" alt="Twitter Logo">
+                </a>
+            </li>
+            <li>
+                <a href="#">
+                    <img id="medium" src="/assets/medium.svg" alt="Medium Logo">
+                </a>
+            </li>
```
- [ ] Very great job while implementing this list of social media, just there is something that we need to improve here.
   - It seems that the icons are displaying on your side but it's not the case on my end, look how it looks like:
      - ![iconb2](https://user-images.githubusercontent.com/57562869/143579932-ae7d4754-51b7-4c3b-bfe8-9523215b31d7.png)
      - This is the problem of the path of images. it seems that you did not put the right path, it doesn't work on my end because Github has case sensitive paths, it's supposed to be very correct.
      - Kindly try to fix that, by fixing that, it will improve the look of your portofolio. :+1:

moise10r commented on Index.html:
```diff
@@ -28,6 +28,204 @@ <h2 id="myTitle">Software Developer</h2>
+    <section class="myAbilities">
+        <img class="line" src="./assets/vectorLine.svg" alt="horizontal line">
+        <div class="skillsSection">        
+            <div class="mySkills">
+                <a id="languages" href="#"><h2>Languages</h2></a>
+                <ul class="abilities">
+                    <li>
+                        <a href="#">JavaScript</a>
+                    </li>
+                    <li>
+                        <a href="#">Ruby</a>
+                    </li>
+                    <li>
+                        <a href="#">Html</a>
+                    </li>
+                    <li>
+                        <a href="#">Html</a>
+                    </li>
+                </ul>
+            </div>
+            <div class="mySkills">
+                <a id="frameworks" href="#"><h2>Frameworks</h2></a>
+                <ul class="abilities">
+                    <li>
+                        <a href="#">Bootstrap</a>
+                    </li>
+                    <li>
+                        <a href="#">Ruby on Rails</a>
+                    </li>
+                    <li>
+                        <a href="#">RSPec</a>
+                    </li>
+                    <li>
+                        <a href="#">CapyBara</a>
+                    </li>
+                    <li>
+                        <a href="#">Selenium</a>
+                    </li>
+                </ul>
+            </div>
```
- [ ] Great job for implementing this list of knowledge, but there is something to check out. As you can see this is actually a list and as it's a list it's supposed to be a `ul` tag instead of a `div`.
   - So for that it should be like this:
   ```HTML
   <ul class="knowledge">
      <li class="languages">
        <ul class="languages-list">
           <li class="language"></li>
           ....
       </ul> 
    </li>
    ....
   </ul>
   ```

moise10r reviewed (changes requested):
### Hi @BenjaminSemah  

Good job so far for implementing this project, it looks really awesome, I appreciate it! :+1: 

## Good points
- No linter errors :heavy_check_mark: 
- proper Github flow :heavy_check_mark: 
- Good PR description :heavy_check_mark: 

There are some issues that you still need to work on to go to the next project but you are almost there!

![require2](https://user-images.githubusercontent.com/57562869/143582101-169896ab-357f-48ec-a87f-271d04f6ea67.gif)

<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

BenjaminSemah committed:
Update markup and stylesheet for Homepage

BenjaminSemah commented:
**Update markup and stylesheet for Homepage**
_In this update, I implemented the following changes;_
- Used flexbox to display sample projects.
- Corrected URL path for social media icons.
- Changed div tag used for knowledge list to ul and li tags.

DammyShittu commented on style.css:
```diff
@@ -67,3 +63,224 @@ header {
+  color: #dcdce8;
+}
+
+#project1 {
+  display: flex;
+  flex-direction: column;
+  background-image: url(./assets/pro1image.svg);
+  background-repeat: no-repeat;
+  background-position: left;
+  padding: 20px;
+}
+
+#project2 {
+  display: flex;
+  flex-direction: column;
+  background-image: url(./assets/pro2image.svg);
+  background-repeat: no-repeat;
+  background-position: right;
+  padding: 20px;
+}
+
+#project3 {
+  display: flex;
+  flex-direction: column;
+  background-image: url(./assets/pro3image.svg);
+  background-repeat: no-repeat;
+  background-position: left;
+  padding: 20px;
+}
+
+#project4 {
+  display: flex;
+  flex-direction: column;
+  background-image: url(./assets/pro4image.svg);
+  background-repeat: no-repeat;
+  background-position: right;
+  padding: 20px;
+}
```
[OPTIONAL]

- [ ] Great implementation of the Flexbox layout suggested by the previous reviewer. 👍
What do you think about following the Don't Repeat Yourself(DRY) rule for the highlighted code? The DRY rule helps to keep the codebase clean and avoid repetition.

To achieve this, you can simply give all four projects the same class name and style that particular style.

Here is an example of how to keep the code above dry. 👇

```
.my-projects {
  display: flex;
  flex-direction: column;
  background-repeat: no-repeat;
  padding: 20px;
}

#project1 {
// other styling
}

#project2 {
// other styling
}

#project3 {
// other styling
}

#project4 {
// other styling
}

```

DammyShittu reviewed (approved):
Hi @BenjaminSemah , 

# APPROVED 🟢 

_Outstanding implementation of the features in this milestone and changes suggested by the previous reviewer.✔️_

You've really done well 👏 

### Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

# CONGRATULATIONS! 🎉 

# OPTIONAL SUGGESTIONS ♻️

- I would recommend you go through this [article](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md) to be updated on the appropriate section to write a PR description. 😀

- Instead of having `<li><img src="./assets/separator.svg" alt="Vertical line"></li>` in multiple places in your HTML code, what do you think about giving the lists that require the vertical line the same class name and then proceed to add some padding and also a border-right declaration in your CSS file? This will help keep the HTML file cleaner and also make it less stressful for you to make a change to the vertical line. 😉 

Kindly check the comment below 👇 

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

# Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
